right/src/usb_interfaces/usb_interface_mouse.c

```diff
@@ -20,25 +20,25 @@ void ResetActiveUsbMouseReport(void)
     bzero(ActiveUsbMouseReport, USB_MOUSE_REPORT_LENGTH);
 }
 
-static volatile usb_status_t usbMouseAction(void)
+usb_status_t usbMouseAction(void)
 {
-    usb_mouse_report_t *mouseReport = getInactiveUsbMouseReport();
-    IsUsbMouseReportSent = true;
-    return USB_DeviceHidSend(UsbCompositeDevice.mouseHandle, USB_MOUSE_ENDPOINT_INDEX,
-               (uint8_t*)mouseReport, USB_MOUSE_REPORT_LENGTH);
+    usb_status_t status = kStatus_USB_Error;
+    if (!IsUsbMouseReportSent) {
+        status = USB_DeviceHidSend(
+                UsbCompositeDevice.mouseHandle, USB_MOUSE_ENDPOINT_INDEX,
+                (uint8_t*)getInactiveUsbMouseReport(), USB_MOUSE_REPORT_LENGTH);
+        IsUsbMouseReportSent = true;
+        UsbMouseActionCounter++;
+    }
+    return status;
 }
 
 usb_status_t UsbMouseCallback(class_handle_t handle, uint32_t event, void *param)
 {
-    UsbMouseActionCounter++;
```

Comment:
@mondalaci not sure if this was put here on purpose? But I moved it into usbMouseAction, as this is how it is done in the other interfaces.

Reply:
It's just a debug variable that is meant to count the invocations of the mouse callback. It's not critical by any means.